src/main/scala/org/mdedetrich/stripe/v1/Plans.scala

```diff
@@ -223,15 +281,26 @@ object Plans extends LazyLogging {
       executionContext: ExecutionContext): Future[Try[Plan]] = {
     val postFormParameters = PostParams.flatten(
       Map(
-        "id"                   -> Option(planInput.id.toString),
-        "amount"               -> Option(planInput.amount.toString()),
-        "currency"             -> Option(planInput.currency.iso.toLowerCase),
-        "interval"             -> Option(planInput.interval.id.toString),
-        "name"                 -> Option(planInput.name),
-        "interval_count"       -> planInput.intervalCount.map(_.toString),
-        "statement_descriptor" -> planInput.statementDescriptor,
-        "trial_period_days"    -> planInput.trialPeriodDays.map(_.toString)
-      )) ++ mapToPostParams(planInput.metadata, "metadata")
+        "id"             -> Option(planInput.id.toString),
+        "amount"         -> Option(planInput.amount.toString()),
+        "currency"       -> Option(planInput.currency.iso.toLowerCase),
+        "interval"       -> Option(planInput.interval.id.toString),
+        "interval_count" -> planInput.intervalCount.map(_.toString),
+        "nickname"       -> planInput.nickname
+      )) ++ mapToPostParams(planInput.metadata, "metadata") ++ {
+      planInput.product match {
+        case service: Product.ServiceProduct =>
+          val params = PostParams.flatten(
+            Map(
+              "id"                   -> service.id,
+              "name"                 -> Option(service.name),
+              "statement_descriptor" -> service.statementDescriptor
+            )
+          )
+          mapToPostParams(Option(params ++ mapToPostParams(service.metadata, "metadata")), "product")
```

Comment:
This was a little bit tricky to work out.
akka-http already includes a [FormData](https://doc.akka.io/api/akka-http/current/akka/http/scaladsl/model/FormData.html) that seems relevant here. Might that make handling all these nested maps a bit simpler?

Reply:
Yeah, we used to use unfiltered and switch to akka http that's why we are not using it.

Personally, I'm not particularly taken with akka http so I would not want to increase our dependency on it.